ui/app/components/transit-key-actions.js

```diff
@@ -35,213 +49,141 @@ const TRANSIT_PARAMS = {
-      if (key === 'signature_algorithm' && !isRSA) {
+      if (key === 'signature_algorithm' && !this.keyIsRSA) {
         return result;
       }
       if (data[key]) {
         result[key] = data[key];
       }
       return result;
     }, {});
-  },
-
-  actions: {
-    onActionChange(action) {
-      set(this, 'selectedAction', action);
-      this.checkAction();
-    },
-
-    onClear() {
-      this.resetParams(null, this.selectedAction);
-    },
-
-    clearParams(params) {
-      const arr = Array.isArray(params) ? params : [params];
-      arr.forEach((param) => this.set(param, null));
-    },
-
-    doSubmit(data, options = {}, maybeEvent) {
-      const event = options.type === 'submit' ? options : maybeEvent;
-      if (event) {
-        event.preventDefault();
+  }
+
+  @action toggleEncodeBase64() {
+    this.props.encodedBase64 = !this.props.encodedBase64;
+  }
+
+  @action clearSpecificProps(arrayToClear) {
+    arrayToClear.forEach((prop) => (this.props[prop] = null));
+  }
+
+  @task
+  @waitFor
+  *doSubmit(data, options = {}, maybeEvent) {
+    const event = options.type === 'submit' ? options : maybeEvent;
+    if (event) {
+      event.preventDefault();
+    }
+    const { backend, id } = this.args.key;
+    const action = this.args.selectedAction || this.firstSupportedAction;
+    const { ...formData } = data || {};
+    if (!this.props.encodedBase64) {
+      if (action === 'encrypt' && !!formData.plaintext) {
+        formData.plaintext = encodeString(formData.plaintext);
       }
-      const { backend, id } = this.getModelInfo();
-      const action = this.selectedAction;
-      const { encodedBase64, ...formData } = data || {};
-      if (!encodedBase64) {
-        if (action === 'encrypt' && !!formData.plaintext) {
-          formData.plaintext = encodeString(formData.plaintext);
-        }
-        if ((action === 'hmac' || action === 'verify' || action === 'sign') && !!formData.input) {
-          formData.input = encodeString(formData.input);
-        }
+      if ((action === 'hmac' || action === 'verify' || action === 'sign') && !!formData.input) {
+        formData.input = encodeString(formData.input);
       }
-      const payload = formData ? this.compactData(formData) : null;
-      this.setProperties({
-        errors: null,
-        result: null,
-      });
-      this.store
+    }
+    const payload = formData ? this.compactData(formData) : null;
+
+    try {
+      const resp = yield this.store
         .adapterFor('transit-key')
-        .keyAction(action, { backend, id, payload }, options)
-        .then(
-          (resp) => this.handleSuccess(resp, options, action),
-          (...errArgs) => this.handleError(...errArgs)
-        );
-    },
-  },
-});
+        .keyAction(action, { backend, id, payload }, options);
+
+      this.handleSuccess(resp, options, action);
+    } catch (e) {
+      this.errors = e.errors;
+    }
+  }
+
+  handleSuccess(resp, options, action) {
```

Comment:
I understand why these were rearranged, but in the future for glimmer PRs can we keep the order as close to the original as possible so the git diff is easier to compare while reviewing?